apps/nestjs-wallet/src/vc-api/exchanges/dtos/custom-validators/presentation-definition-credential-query.validator.ts

```diff
@@ -0,0 +1,35 @@
+import { IPresentationDefinition, PEX, Status } from '@sphereon/pex';
+import { registerDecorator, ValidationOptions, ValidationArguments } from 'class-validator';
+
+/**
+ * https://github.com/typestack/class-validator#custom-validation-decorators
+ */
+export function IsPresentationDefinitionCredentialQuery(validationOptions?: ValidationOptions) {
+  return function (object: Object, propertyName: string) {
+    registerDecorator({
+      name: 'isPresentationDefinitionCredentialQuery',
+      target: object.constructor,
+      propertyName: propertyName,
+      constraints: [],
+      options: validationOptions,
+      validator: {
+        validate(value: IPresentationDefinition, args: ValidationArguments) {
+          const pex = new PEX();
+          const validated = pex.validateDefinition(value);
+          if (Array.isArray(validated)) {
+            validated.forEach((checked) => {
+              if (checked.status === Status.ERROR || checked.status === Status.WARN) {
+                return false;
```

Comment:
I don’t think it’s gone work as you wished. This is a return statements for a ‘forEach’ method. It looks like for me that’s array will always be valid object in this case.

Reply:
Lol, nice catch, thanks 🤦